File: CPP/POO_CPP/MundoAnimal/Gato.cpp
#include <iostream>
#include "Animal.h"
#include "Gato.h"

using namespace std;

Gato::Gato(string nome): Animal(nome) {
  Gato::ct++;
}

Gato::~Gato() {
}

void Gato::som() {
  cout << "Miauuuuuuuuuu!! Miauuuuuuuuu!!" << endl;
}

string Gato::toString() {
  return "Gato{" + getNome() + '}';
}

File: POO_2026_1/src/POO/Gato.java
package POO;

public class Gato extends Animal {
    public static int ct = 0;

    public Gato(String nome) {
        super(nome);
        ct = ct + 1;
    }   

    @Override
    public void som() {
      System.out.println("Miauuuuuuu!!! Miauuuuuuuu!!! Miauuuuuuu!!!");  
    }

    @Override
    public String toString() {
        return "Gato{" + super.getNome() + '}';
    }

}
